examples/tls-logger/server.cpp

```diff
@@ -136,14 +145,18 @@ class server
     server(
         const std::string& cert_file, 
         const std::string& key_file,
-        const uint16_t port) : _ctx(
+        const uint16_t port,
+	uint8_t num_threads) : _ctx(
             ice::SERVER_TCP_SOCKET,
             cert_file,
             key_file)
+	    
     {
         _srv = common::open_socket(common::SERVER, nullptr, port);
         _file_path = "output_file.txt";
         std::fstream file_stream(_file_path, std::ios::out);
+	threads = num_threads;
+        thread_list.resize(threads);
```

Comment:
It might be easier to use `reserve()` instead of `resize()`. The difference is that `resize()` will call the `std::thread()` constructor, which is not necessary. It is however good practice to allocate the memory in the `std::vector` once you know how large it is supposed to be, and calling `reserve()` would do that.